scripts/grype_report_parser.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type VulnerabilityScannerResponse struct {
+	Matches                json.RawMessage `json:"matches"`
+	RelatedVulnerabilities json.RawMessage `json:"relatedVulnerabilities"`
+	MatchDetails           json.RawMessage `json:"matchDetails"`
+	Artifact               json.RawMessage `json:"artifact"`
+}
+
+type MatchResponse struct {
+	Vulnerabilities VulnerabilityReport `json:"vulnerability"`
+}
+
+type FixVersionsResponse struct {
+	Versions []string `json:"versions"`
+	State    string   `json:"state"`
+}
+
+type VulnerabilityReport struct {
+	Id          string              `json: id`
+	Severity    string              `json:"severity"`
+	Namespace   string              `json:"namespace"`
+	Description string              `json:"description"`
+	FixVersions FixVersionsResponse `json:"fix"`
+}
+
+// Filters vulnerabilites based on the severity levels set in severityTypeSet
+func filterVulnerabilityReportMatches(matches []MatchResponse, severityTypeSet map[string]bool) ([]VulnerabilityReport, error) {
+	matchingVulnerabilities := make([]VulnerabilityReport, 0)
+	for _, match := range matches {
+		if severityTypeSet[match.Vulnerabilities.Severity] {
+			matchingVulnerabilities = append(matchingVulnerabilities, match.Vulnerabilities)
+		}
+	}
+	return matchingVulnerabilities, nil
+}
+
+// Unmarshalls the Matches from the vulnerability report and returns a list of vulnerabilities
+// based on the severity levels set in severityTypeSet
+func decodeVulnerabilityReports(vulnerabilityScannerResponse VulnerabilityScannerResponse, severityTypeSet map[string]bool) ([]VulnerabilityReport, error) {
+	var matches []MatchResponse
+	matchingVulnerabilities := make([]VulnerabilityReport, 0)
+	if err := json.Unmarshal(vulnerabilityScannerResponse.Matches, &matches); err != nil {
+		return matchingVulnerabilities, errors.New("Error while unmarshalling a MatchResponse with err: " + err.Error())
+	}
+	return filterVulnerabilityReportMatches(matches, severityTypeSet)
+}
+
+// Unmarshalls the MatchDetails from the vulnerability report and returns a list of vulnerabilities
```

Comment:
```suggestion
// parseVulerabilitiesReport unmarshals the MatchDetails from the vulnerability report and returns a list of vulnerabilities
```
[typo] Just a [nit]. I think it's a single `l` in `unmarshal`

Reply:
Also, it's not clear what `MatchDetails` is

Reply:
Fixed the spelling issue. 
MatchDetails is one of the keys in the JSON result returned by Grype. Hence, I agree that mentioning it without adding the context of what the JSON entails adds to the confusion. Hence, I've changed the documentation into a more generic comment about the behavior of the function.